docs/source/deploying/workers.rst

```diff
@@ -0,0 +1,58 @@
+Task Workers
+=============
+
+Quetz offers 3 types of backends for task workers. Each of them is explained below.
+
+Thread
+-----------
+
+Thread workers process tasks in a separate thread. This functionality is in-built into FastAPI using
+``BackgroundTasks`` (https://fastapi.tiangolo.com/tutorial/background-tasks/).
+
+
+Subprocess
+-----------
+
+Subprocess workers start in a separate process and are implemented through ``ProcessPoolExecutor`` of the
+``concurrent.futures`` module. Once again, this is shipped as a part of Quetz.
+
+
+Redis
+-----------
+
+For advanced use-cases, Quetz also offers the ability to use ``redis-queue`` to manage jobs and run them on
+multiple processes or even multiple servers.
+
+To use this backend, one needs to setup ``redis`` and ``redis-queue``.
+
+Setting up ``redis``
+^^^^^^^^^^^^^^^^^^^^
+
+Make sure that ``redis`` is installed. There are multiple ways to do this. One can compile it from source (recommended),
+use a package manager for your distribution (such as ``brew`` for MacOS) and even use a Docker image.
+
+Once ``redis`` is installed, it needs to be started. This is as simple as executing the command ``redis-server`` on a
+terminal (or can be run in a container).
+
+We also need to install ``redis-py`` (https://github.com/andymccurdy/redis-py) - the python client for Redis.
+
+Installing ``redis-queue``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``redis-queue`` is a python library that facilitates using Redis for queueing jobs and processing them in the background with
+workers. The installation can be done by following instructions here: https://python-rq.org/#installation
```

Comment:
```suggestion
workers. The installation can be done by following `instructions`_.

.. instructions: https://python-rq.org/#installation
```

you need to add urls explicitly in sphinx (as in the example above): https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#external-links

Reply:
Thanks